src/internet_identity/tests/integration/config.rs

```diff
@@ -2,189 +2,216 @@ use canister_tests::api::internet_identity as api;
 use canister_tests::framework::{
     env, install_ii_canister_with_arg, upgrade_ii_canister_with_arg, II_WASM,
 };
+use ic_cdk::api::management_canister::provisional::CanisterId;
 use internet_identity_interface::internet_identity::types::{
     ArchiveConfig, CaptchaConfig, CaptchaTrigger, InternetIdentityInit, OpenIdConfig,
     RateLimitConfig,
 };
-use pocket_ic::CallError;
+use pocket_ic::{CallError, PocketIc};
+
+const DEFAULT_CONFIG: InternetIdentityInit = InternetIdentityInit {
+    assigned_user_number_range: Some((10000, 1_000_000)),
+    archive_config: Some(ArchiveConfig {
+        module_hash: [17; 32],
+        entries_buffer_limit: 123789,
+        polling_interval_ns: 659871258,
+        entries_fetch_limit: 33,
+    }),
+    canister_creation_cycles_cost: Some(123),
+    register_rate_limit: Some(RateLimitConfig {
+        time_per_token_ns: 99,
+        max_tokens: 874,
+    }),
+    captcha_config: Some(CaptchaConfig {
+        max_unsolved_captchas: 788,
+        captcha_trigger: CaptchaTrigger::Dynamic {
+            threshold_pct: 12,
+            current_rate_sampling_interval_s: 456,
+            reference_rate_sampling_interval_s: 9999,
+        },
+    }),
+    related_origins: None,
+    openid_google: None,
+};
+
+/// Utility method to check if install/upgrade config matches config returned from query,
+/// this maps `None` to `Some(None)` for `Option<Option>` since it does not return `None`.
+fn assert_config(
+    env: &PocketIc,
+    canister_id: CanisterId,
+    expected_config: &InternetIdentityInit,
+) -> Result<(), CallError> {
+    assert_eq!(
+        api::config(env, canister_id)?,
+        InternetIdentityInit {
+            openid_google: expected_config.openid_google.clone().or(Some(None)),
+            ..expected_config.clone()
+        }
+    );
+    Ok(())
+}
+
+/// Utility method to install and check if config matches install config
+fn install_and_assert(
+    env: &PocketIc,
+    config: &InternetIdentityInit,
+) -> Result<CanisterId, CallError> {
+    let canister_id = install_ii_canister_with_arg(env, II_WASM.clone(), Some(config.clone()));
+    assert_config(env, canister_id, config)?;
+    Ok(canister_id)
+}
+
+/// Utility method to upgrade and check if config matches upgrade config
+fn upgrade_and_assert(
+    env: &PocketIc,
+    canister_id: CanisterId,
+    config: &InternetIdentityInit,
+    expected_config: &InternetIdentityInit,
+) -> Result<(), CallError> {
+    upgrade_ii_canister_with_arg(env, canister_id, II_WASM.clone(), Some(config.clone()))?;
+    assert_config(env, canister_id, expected_config)?;
+    Ok(())
+}
+
+/// Utility method to upgrade with other config change and check if any other config hasn't changed
+fn unrelated_change_and_assert(
```

Comment:
`unrelated_change` to what? To another config? Which other config? One with optionals?